File: Distributed_Systems/ArgsAndInput/files/ArgsAndInputDemo.java
import java.io.InputStreamReader;
import java.io.BufferedReader;

public class ArgsAndInputDemo {

	public static void main(String[] args) {
		// read the command line args
		if ( args.length < 2 )
		{
			System.err.println( "ArgsAndInputDemo requires 2 command line arguments" );
			return;
		}
		System.out.println( "Arg[0] is " + args[0] );
		System.out.println( "Arg[1] is " + args[1] );
		
		// read standard input
		try	{
			// input the message from standard input
			BufferedReader input= new BufferedReader( 
			  new InputStreamReader(System.in) );
			String line;
			while( (line= input.readLine()) != null )				
				System.out.println( "You entered: " + line );			
		}
		catch ( Exception e ) {
			System.out.println( e.getMessage() );
		}
	}
}

File: Distributed_Systems/HW1 Args And Input/args-and-input-demo.py
#! python

# demo of command line arguments and standard input handling

# get the command line arguments
import sys
argc= len( sys.argv )
print( 'There are ' + str(argc) + ' command line arguments' )
for arg in sys.argv:
	print( arg )
# read from standard input
#text= input() # requires exception processing!
text= sys.stdin.readline() # leaves the newline char at the end
lines= []
while text:
	lines.append( text )
	#text= input() # requires exception processing!
	text= sys.stdin.readline()
for line in lines:
	print( line, end= '' )
